polkadot-protocol/glossary.md

```diff
@@ -56,6 +56,10 @@ An early approach to Byzantine fault tolerance (BFT), practical Byzantine fault
 
 The communication overhead for such systems is `O(n²)`, where `n` is the number of nodes (participants) in the system.
 
+### Preimage
+
+A preimage is the data that is input into a hash function to calculate a hash. Since a hash function is a one-way function, the output, the hash, cannot be used to reveal the input, the preimage.
```

Comment:
WDYT about adding the following reference?

```suggestion
A preimage is the data that is input into a hash function to calculate a hash. Since a hash function is a [one-way function](https://en.wikipedia.org/wiki/One-way_function){target=\_blank}, the output, the hash, cannot be used to reveal the input, the preimage.
```